src/spikeinterface/comparison/comparisontools.py

```diff
@@ -85,27 +108,283 @@ def do_count_event(sorting):
     return event_counts
 
 
-def count_match_spikes(times1, all_times2, delta_frames):  # , event_counts1, event_counts2  unit2_ids,
+def get_optimized_compute_dot_product_all():
+    """
+    This function wraps around the compute_dot_product function,
+    which uses numba for JIT compilation. It caches the compiled function
+    for improved performance on subsequent calls.
     """
-    Computes matching spikes between one spike train and a list of others.
 
-    Parameters
-    ----------
-    times1: array
-        Spike train 1 frames
-    all_times2: list of array
-        List of spike trains from sorting 2
+    if hasattr(get_optimized_compute_dot_product_all, "_cached_function"):
+        return get_optimized_compute_dot_product_all._cached_function
 
-    Returns
-    -------
-    matching_events_count: list
-        List of counts of matching events
+    import numba
+
+    @numba.jit(nopython=True, nogil=True)
+    def compute_dot_product(sample_frames, unit_indices, num_units_sorting1, num_units_sorting2, delta_frames):
+        """
+        Compute the dot product matrix for two spike trains.
+
+        Note that the sample frames and unit indices must be sorted by sample frames.
+        Also note that the unit_indices corresponding to the second spike train must be
+        shifted by num_units_sorting1.
+
+        This creates a matrix that can be divided in four quadrants. Clokwise from top left:
+        1. The dot product of spikes in the first spike train with themselves.
+        2. The dot product of spikes in the first spike train with spikes in the second spike train.
+        3. The dot product of spikes in the second spike train with spikes in the first spike train.
+        4. The dot product of spikes in the second spike train with themselves.
+
+        Note that the norms of of the spike trains are the diagonals of the 1 and 4 quadrants.
```

Comment:
```suggestion
        Note that the norms of the spike trains are the diagonals of the 1st and 4th quadrants.
```

Reply:
Thanks @zm711 . I will erase this function though. After profiling turns out that computing norms and dot products separated is as computational efficient and it is way more flexible.

Reply:
No worries. I'm going to be off GH for about the next week, so I just wanted to make sure I didn't miss active things while I had the chance.

Reply:
Thanks for all the help though. Good luck in the week ahead : )